lib/request.js

```diff
@@ -48,6 +60,7 @@ module.exports = function (protocol, host, version, token) {
         object.url = getUrl(object);
         delete object.endpoint;
         object.json = object.json || true;
+        object.timeout = timeout;
```

Comment:
optional timeout - can be undefined, request just treats it like it's not specified at all